R/utils-json.R

```diff
@@ -193,7 +193,13 @@ getApprovals <- function(data, chain_nm, legend_nm, appr_var_all, month=NULL, po
     for(i in seq_along(date_index_list)){
       d <- date_index_list[[i]]
       
-      applicable_dates <- points[['time']][d]
+      title <- getReportMetadata(data, 'title')
+      #result <- !objectName %in% listObjects
+      if (grepl(title, 'DV Hydrograph')) {
+        applicable_dates <- points[['time']][d] + hours(23) + minutes(59)
```

Comment:
This DV specific logic should be moved up and out of the function. The function should remain generic and should not have any hidden magic gnome logic.

To do this, wherever DV hydrograph calls this function, that should be hte place that adds the hours/minutes back in if needed.